packages/build/src/tracing/main.ts

```diff
@@ -0,0 +1,79 @@
+import { HoneycombSDK } from '@honeycombio/opentelemetry-node'
+import { context, trace, propagation, SpanStatusCode } from '@opentelemetry/api'
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+import { NodeSDK } from '@opentelemetry/sdk-node'
+
+import type { TracingOptions } from '../core/types.js'
+import { ROOT_PACKAGE_JSON } from '../utils/json.js'
+
+let sdk: NodeSDK
+
+/** Starts the tracing SDK, if there's already a tracing service this will be a no-op */
+export const startTracing = function (options: TracingOptions) {
+  if (!options.enabled) return
+  if (sdk) return
+
+  sdk = new HoneycombSDK({
```

Comment:
I assume it will use the API key via the env var `HONEYCOMB_API_KEY`. Is this set when we run the build script?

We should also make sure the key has limited permissions, probably only `create events`.

Reply:
> I assume it will use the API key via the env var HONEYCOMB_API_KEY.

No need, since we're using the collector this is all handled for us, we just need to ship it to the specific host.

- https://github.com/netlify/buildbot/blob/c8284ab508ec067beb7d47e5b235c223b4beb34f/_infra/terraform/buildbot-provisioning/helm/otel-collector.yaml.tpl#L16-L20
- https://github.com/netlify/buildbot/blob/c8284ab508ec067beb7d47e5b235c223b4beb34f/script/entrypoint.sh#L4-L10
- https://github.com/netlify/buildbot/blob/c8284ab508ec067beb7d47e5b235c223b4beb34f/script/run-build.sh#L113-L117